src/python/pants/core/goals/check_test.py

```diff
@@ -68,7 +69,10 @@ def check_results(self) -> CheckResults:
 
 class SuccessfulRequest(MockCheckRequest):
     tool_name = "Successful Checker"
-    tool_id = "successfulchecker"
+
+    @classproperty
+    def tool_id(cls) -> str:
+        return "successfulchecker"
```

Comment:
what drived this change to use `@classproperty` over `tool_id: ClassVar[str] = "..."` ? 

(preferring `ClassVar` over `@classproperty` when possible as "not invented here") ;)

Reply:
It's internal and it's derived from other class properties, so I can't use a class variable :/